### PR TITLE
Transfers mind on brain replacement

### LINKED ISSUE #26078: Using autosurgeon to replace brains
Using autosurgeon with a target's brain results in the brain's player being put in control instead.

That's easily a feature, but the problem with it is that the brain that drops on the floor as a result of the procedure is not the brain of the autosurgeon user, `John Doe's brain` that John Doe can re-enter after ghosting out.

Instead, it's just a playerless `brain` and John Doe has no body to re-enter once his ghost leaves the brain

### PR DESCRIPTION
**Pull request by QualityVan**
Fixes #26078 , the brain dropped from the body now has the old inhabitant.

### CODE REVIEW

File: code/modules/mob/living/brain/brain_item.dm [hunk 1]
```diff
@@ -37,7 +37,7 @@
 		if(C.has_brain_worms())
 			var/mob/living/simple_animal/borer/B = C.has_brain_worms()
 			B.leave_victim() //Should remove borer if the brain is removed - RR
-		transfer_identity(C)
+	transfer_identity(C)
```
Comment (RemieRichards) [inline code/modules/mob/living/brain/brain_item.dm]: pretty sure the ```special``` is how changelings get away with dropping fake brains without BECOMING those brains.

Comment (QualityVan) [inline code/modules/mob/living/brain/brain_item.dm]: Fun fact: according to brief tests, surgical brain removal takes the real brain from changelings on live. Dismemberment uses transfer_to_limb, which does explicitly check for changeling status and uses a fake brain if it's the case. I could add a fix for the first bit if desired.

Comment (RemieRichards) [inline code/modules/mob/living/brain/brain_item.dm]: Yeah it should take the fake one, surgery wise, as the real changeling's brain is "in the chest" (it's not but it's certainly not the head) so being able to do surgery on the head and getting the real one is... weird and wrong.

Comment (QualityVan) [inline code/modules/mob/living/brain/brain_item.dm]: Should that be in a separate pull or is atomization more a thing for features? Also, fixing surgical removal would allow a changeling test by cutting out the brain: the examine tests whether the individual has a brain or not, so when a changeling keeps their brain and gets examined they'll clearly not be brainless.

Comment (RemieRichards) [inline code/modules/mob/living/brain/brain_item.dm]: well you'd have to fake the examine too, I suppose.
Also yes it should be its own PR.

**Comment by QualityVan** [comment]
Closing due to potential conflicts with changeling stuff.
